### ISSUE TITLE
Offset block placement broken

### ISSUE THREAD

**Issue by end-user** [maruohon/tweakeroo#338]
In v0.13.1, placing blocks using `flexibleBlockPlacementOffset` while clicking in the center region (that is, skipping a block) results in ghost blocks.

**Comment by mazen** [comment]
I have the same issue with the placement rotations.

client: 
tweakeroo-fabric-1.18.2-0.13.1.jar with the carpetAccuratePlacementProtocol set to true

server:
carpet-extra-1.18.2-pre2-1.4.64
fabric-carpet-1.18.2-1.4.66+v220228

**Comment by maruohon** [comment]
There was an added vanilla anti-cheat/whatever check, which basically broke the accurate placement protocol. I sent a PR to fix that to CarpetExtra, but until that gets merged, I have the fixed version of CarpetExtra built here:
https://cdn.discordapp.com/attachments/775717305522192424/952634690383253664/carpet-extra-1.18.2-1.4.64-masa.2.jar

You need this on the server side. Tweakeroo 0.13.1 contains the same fix for the client side and single player.

**Comment by spiiross** [comment]
Just a heads-up @maruohon ,
If you play `single player world` with carpetExtra and tweakeroo ,enable the carpetExtra rule  `accurateBlockPlacement ` . It creates ghost blocks if it's disabled. If you don't use any of carpetExtra features  then simply remove the jar file from the mod folder, either way fixes the issue.

**Comment by Hi30MC** [comment]
> There was an added vanilla anti-cheat/whatever check, which basically broke the accurate placement protocol. I sent a PR to fix that to CarpetExtra, but until that gets merged, I have the fixed version of CarpetExtra built here: https://cdn.discordapp.com/attachments/775717305522192424/952634690383253664/carpet-extra-1.18.2-1.4.64-masa.2.jar
> 
> You need this on the server side. Tweakeroo 0.13.1 contains the same fix for the client side and single player.

Would there be a way to apply this to the client-side such that there isn't a need to have CarpetExtra serverside? I am playing on a paper server that has no intention to switch to fabric, and it is a tad annoying that I can no longer place blocks in the center region.

**Comment by maruohon** [comment]
@HiHi3690 There is no way to apply the fix on the client only when playing on a server, if you want to use block rotations via the accurate placement protocol, because using the protocol at all will "almost by definition" trigger the vanilla check.

However the offset placement not working anymore with the center region is a separate bug where the mod unintentionally sends the wrong hit position. So that can be fixed so that it sends the "normal" hit position and doesn't trigger the vanilla check. However all the placement related code is such a huge mess currently that I may need to do the long-planned total rewrite of that code to be able to fix it...

**Comment by Hi30MC** [comment]
Yeah, I figured that the block rotations would break almost instantly, which I have encountered in the past, and I was only wondering about the offset placement. I am glad to hear that it may just be a jank code issue, rather than a vanilla anti-cheat issue. Thanks for the response!
